packages/devtools_app/lib/src/timeline/flutter_frames_chart.dart

```diff
@@ -198,6 +207,20 @@ class _FlutterFramesChartState extends State<FlutterFramesChart>
       ],
     );
   }
+
+  Widget _buildAverageFps() {
+    final double sumFrameTimesMs = widget.frames.fold(
+        0.0,
+        (sum, frame) =>
+            sum + math.max(frame.uiDurationMs, frame.rasterDurationMs));
```

Comment:
if you take the max of
math.max(frame.uiDurationMs, frame.rasterDurationMs, 1000/averageFrameTime)
you will never pretend a frame ran faster than the target frame time. That will also ensure we never report above the target frame rate and ensure that we don't claim you hit 60fps even though you had one frame that took 400ms and then a bunch of frames that were very quick.

Reply:
What about the scenario where you have 1000 frames that run in "16.6 ms" (adjusted for this calculation) and a single frame that runs in 18 ms. Then the "average" FPS would be slower than 60 even though one frame barely missed the budget. Perhaps we should remove the word "average" from the label if we are going to adjust times in the calculation?

Reply:
If one frame missed the budget I'm happy if the frame rate is slightly bellow 60fps and the app didn't hit 60fps.
Another way to think about it is
One frame at 18ms and one frame at 14ms is a lot worse than
Two frames both at 16ms.

Reply:
I see. Addressed your comments.